sdk/cosmos/azure-cosmos/docs/ErrorCodesAndRetries.md

```diff
@@ -0,0 +1,18 @@
+| 404/1002    | <ul><li>For write operations: </br><ul><li>If multiple write locations are enabled for the account, the SDK will fetch the write endpoints and retry once per each of these. </li><li>The client refreshes it's locational endpoints and retries requests when the user has enabled endpoint discovery in their client (default behavior).</li><li>If the account does not have multiple write locations enabled, the SDK will retry only once. </li></ul></li><li>For read operations: </br><ul><li>If multiple write locations are enabled for the account, the SDK will fetch the read endpoints and retry once per each of these. </li><li>The client refreshes it's locational endpoints and retries requests when the user has enabled endpoint discovery in their client (default behavior).</li><li>If the account does not have multiple write locations enabled, the SDK will retry only once. </li>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
+| 408    | <ul><li>For Write Operations: <br><ul><li>Timeout exceptions can be encountered by both the client as well as the server. </li><li>Server-side timeout exceptions are not retried for write operations as it is not possible to determine if the write was in fact successfully committed on the server. To avoid overwriting a previously committed write, the client does NOT retry this operation and bubbles up the exception to the application as a Request Timeout Exception (408). </li><li>For a client-generated timeout exception, there are two scenarios: </br><ul><li>The request was sent over the wire to the server by the client, but the network request timeout exceeded, while waiting for a response. In this case, it is unclear if the write was received and committed by the server and thus, the operation is not retried. A timeout exception is bubbled up to the application by the client. </li><li>The request was not sent over the wire to the server which resulted in a client-generated timeout. The client does NOT retry.</li></ul></li><li>For Query and Point Read Operations:</br><ul><li>The request is retried locally for up to 120 attempts with exponential backoff. </li><li>The SDK will mark the current endpoint as unavailable, refresh its location based routing directives, and set the new endpoint before retrying.</li></ul> </li></ul> |
```

Comment:
i see, so in python, even if the request is not sent through the wire, SDK is not going to retry

Reply:
I do not think in Java and .net, we are marking region as unavailable based on http timeout, as it is not a strong signal to say the region is down